pkg/gardenadm/cmd/connect/connect.go

```diff
@@ -46,7 +64,111 @@ gardenadm connect`,
 	return cmd
 }
 
-func run(_ context.Context, opts *Options) error {
-	opts.Log.Info("Command is work in progress")
+func run(ctx context.Context, opts *Options) error {
+	opts.Log.Info("Using resources from directory", "configDir", opts.ConfigDir)
+
+	b, err := botanist.NewAutonomousBotanistFromManifests(ctx, opts.Log, nil, opts.ConfigDir, true)
+	if err != nil {
+		return fmt.Errorf("failed creating autonomous botanist: %w", err)
+	}
+	b.SeedClientSet, err = b.CreateClientSet(ctx)
+	if err != nil {
+		return fmt.Errorf("failed creating client set for autonomous shoot: %w", err)
+	}
+
+	gardenClientSet, err := kubernetes.NewWithConfig(kubernetes.WithRESTConfig(&rest.Config{
+		Host:            opts.ControlPlaneAddress,
+		TLSClientConfig: rest.TLSClientConfig{CAData: opts.CertificateAuthority},
+		BearerToken:     opts.BootstrapToken,
+	}), kubernetes.WithClientOptions(client.Options{Scheme: kubernetes.GardenScheme}), kubernetes.WithDisabledCachedClient())
+	if err != nil {
+		return fmt.Errorf("failed creating garden client set: %w", err)
+	}
+	b.GardenClient = gardenClientSet.Client()
+
+	gardenletDeployer := newGardenletDeployer(b, gardenClientSet)
+
+	var (
+		g = flow.NewGraph("connect")
+
+		_ = g.Add(flow.Task{
+			Name: "Deploying gardenlet into autonomous shoot cluster",
+			Fn: func(ctx context.Context) error {
+				_, err := gardenletDeployer.Reconcile(
+					ctx,
+					b.Logger,
+					b.Shoot.GetInfo(),
+					nil,
+					&seedmanagementv1alpha1.GardenletDeployment{},
+					&runtime.RawExtension{Object: &gardenletconfigv1alpha1.GardenletConfiguration{}},
+					seedmanagementv1alpha1.BootstrapToken,
+					false,
+				)
+				return err
+			},
+		})
```

Comment:
Do we want to add a check that the `gardenlet` deployment was actually successful, i.e. some health check, or is this only useful once `gardenlet` also performs some tasks?
WDYT?